tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/Path.java

```diff
@@ -71,10 +71,64 @@ public Path(String device, String measurement) {
     }
     this.device = device;
     this.measurement = measurement;
-    this.fullPath = device + TsFileConstant.PATH_SEPARATOR + measurement;
+    this.fullPath = device + TsFileConstant.PATH_SEPARATOR + (measurement.contains(TsFileConstant.PATH_SEPARATOR) ? "\"" + measurement + "\"": measurement);
+  }
+
+  /**
+   * extract device and measurement info from complete path string
+   *
+   * @param pathSc complete path string
+   * @return str[0] is the device string, and str[1] is the measurement string
+   */
+  private void init(String pathSc) {
+    pathSc = pathSc.trim();
+    int i = pathSc.length() - pathSc.replace("\"", "").length();
+    int j = pathSc.length() - pathSc.replace("\'", "").length();
+    if ((i != 2 && i != 0) || (j != 2 && j != 0)) {
+      throw new IllegalArgumentException("input pathSc single/double quotes error, not in pair or more than one pair!");
+    }
+    if ((i == 2 && pathSc.length() - 1 != pathSc.lastIndexOf("\""))
+            || (j == 2 && pathSc.length() - 1 != pathSc.lastIndexOf("\'"))) {
+      throw new IllegalArgumentException("input pathSc contains quoted string in the middle!");
+    }
+    String[] subStrs;
+    if (i != 0 || j != 0) {
+      if (i == 2) {
+        subStrs = pathSc.split("\"");
+      } else {
+        subStrs = pathSc.split("\'");
+      }
+      if(subStrs.length < 2){
+        device = "";
+        fullPath = measurement = subStrs[1];
+      }else {
+        device = subStrs[0];
+        if(!device.trim().equals("")){
+          device = device.substring(0, subStrs[0].length()-1);
+        }
+        measurement = subStrs[1];
+      }
+      fullPath = pathSc;
+    } else {
+      StringContainer sc = new StringContainer(pathSc.split(TsFileConstant.PATH_SEPARATER_NO_REGEX), TsFileConstant.PATH_SEPARATOR);
+      if (sc.size() <= 1) {
+        device = "";
+        fullPath = measurement = sc.toString();
+      } else {
+        device = sc.getSubStringContainer(0, -2).toString();
+        measurement = sc.getSubString(-1);
+        fullPath = sc.toString();
+      }
+    }
   }
 
   public static Path mergePath(Path prefix, Path suffix) {
+    if(suffix.fullPath == "") {
+      return prefix;
+    }
+    else if(prefix.fullPath == "") {
```

Comment:
Use `equals()` to compare strings. And better to put the constants as the left operands.

Reply:
Thanks for your suggestion! The equals() method is used now.